client/src/components/TodoList/AddTodoForm.js

```diff
@@ -1,28 +1,48 @@
-import React, {PropTypes, Component} from 'react'
-
-class AddTodoForm extends Component {
-  render() {
-    let title = ''
-    const {store} = this.context
-    const state = store.getState()
-    return (
-      <div className="todo-add-form">
-        <form onSubmit={e => {
-          e.preventDefault()
-          this.props.onAdd(title.value)
-          title.value = ''
-        }}>
-          <input placeholder="Add todo here..."
-                 ref={input => title = input}/>
-          <button>Add Todo</button>
-        </form>
-      </div>
-    )
-  }
-}
-
-AddTodoForm.contextTypes = {
-  store: PropTypes.object
-}
-
-export default AddTodoForm
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { v4 } from 'uuid';
+
+import { addTodoAction } from '../../redux/actions';
+
+const AddTodoFormComponent = ({ addTodo }) => {
+  const [title, setTitle] = useState('');
+
+  const onSubmit = e => {
+    e.preventDefault();
+    const id = v4();
+
+    addTodo({
+      id,
+      title,
+    });
+
+    setTitle('');
+
+    fetch('/api/todos', {
+      method: 'POST',
+      body: JSON.stringify({ title }),
+      headers: { 'Content-Type': 'application/json' },
+    }).then(response => response.json());
+  };
+
+  return (
+    <div className="todo-add-form">
+      <form onSubmit={onSubmit}>
+        <input
+          placeholder="Add todo here..."
+          value={title}
+          onChange={e => setTitle(e.target.value)}
+        />
+        <button type="button">Add Todo</button>
```

Comment:
```suggestion
        <button>Add Todo</button>
```
Otherwise, form would not submit upon click on the button.

Reply:
As far as I remember, eslint went nuts about _button has to have button type_ 
Anyways, it probably needs the `submit` type, so it would trigger form's `onSubmit`

Reply:
Could you please try it?

Reply:
It took me 3 days, but I did it. 😎

Reply:
Finally, thanks🙂